src/Symfony/Installer/NewCommand.php

```diff
@@ -182,6 +182,15 @@ protected function checkSymfonyVersionIsInstallable()
             ));
         }
 
+        if (preg_match('/^(\d\.\d).*-(dev|beta)\d*$/', $this->version, $matches)) {
```

Comment:
I think the regular expression should be `/^(\d\.\d).*-(dev|beta\d*)$/` instead. Usually there will be no number behind the `-dev` suffix. And what about making the match case-insensitive to support version numbers like `2.7-BETA`?